```
2022-03-07T17:08:31.3834780Z === RUN   TestHealingDanglingObject
2022-03-07T17:08:31.3988082Z Formatting 1st pool, 1 set(s), 16 drives per set.
2022-03-07T17:08:31.4027610Z WARNING: Host local has more than 4 drives of set. A host failure will result in data becoming unavailable.
2022-03-07T17:08:31.6914144Z     erasure-healing_test.go:497: Expected versions 3, got 2
2022-03-07T17:08:31.7002930Z --- FAIL: TestHealingDanglingObject (0.32s)
```
I will have to check if this is a test problem of real before we can merge.

--- a/cmd/erasure-healing_test.go
+++ b/cmd/erasure-healing_test.go
@@ -450,14 +450,17 @@ func TestHealingDanglingObject(t *testing.T) {
 		t.Fatalf("Expected versions 1, got %d", fileInfoPreHeal.NumVersions)
 	}
 
+	ch := make(chan struct{})
 	if err = objLayer.HealObjects(ctx, bucket, "", madmin.HealOpts{Remove: true},
 		func(bucket, object, vid string) error {
 			_, err := objLayer.HealObject(ctx, bucket, object, vid, madmin.HealOpts{Remove: true})
+			close(ch)
 			return err
 		}); err != nil {
 		t.Fatal(err)
 	}
 
+	<-ch
 	fileInfoPostHeal, err = disks[0].ReadVersion(context.Background(), bucket, object, "", false)
 	if err != nil {
 		t.Fatal(err)

--- a/cmd/erasure-healing_test.go
+++ b/cmd/erasure-healing_test.go
@@ -382,6 +382,13 @@ func TestHealingDanglingObject(t *testing.T) {
 		copy(disks, newDisks)
 		objLayer.(*erasureServerPools).serverPools[0].erasureDisksMu.Unlock()
 	}
+	getDisk := func(n int) StorageAPI {
+		objLayer.(*erasureServerPools).serverPools[0].erasureDisksMu.Lock()
+		disk := disks[n]
+		objLayer.(*erasureServerPools).serverPools[0].erasureDisksMu.Unlock()
+		return disk
+	}
+
 	// Remove 4 disks.
 	setDisks(nil, nil, nil, nil)
 
@@ -440,8 +447,8 @@ func TestHealingDanglingObject(t *testing.T) {
 	}
 
 	setDisks(orgDisks[:4]...)
-
-	fileInfoPreHeal, err = disks[0].ReadVersion(context.Background(), bucket, object, "", false)
+	disk := getDisk(0)
+	fileInfoPreHeal, err = disk.ReadVersion(context.Background(), bucket, object, "", false)
 	if err != nil {
 		t.Fatal(err)
 	}
@@ -458,7 +465,8 @@ func TestHealingDanglingObject(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	fileInfoPostHeal, err = disks[0].ReadVersion(context.Background(), bucket, object, "", false)
+	disk = getDisk(0)
+	fileInfoPostHeal, err = disk.ReadVersion(context.Background(), bucket, object, "", false)
 	if err != nil {
 		t.Fatal(err)
 	}
@@ -488,7 +496,8 @@ func TestHealingDanglingObject(t *testing.T) {
 
 	setDisks(orgDisks[:4]...)
 
-	fileInfoPreHeal, err = disks[0].ReadVersion(context.Background(), bucket, object, "", false)
+	disk = getDisk(0)
+	fileInfoPreHeal, err = disk.ReadVersion(context.Background(), bucket, object, "", false)
 	if err != nil {
 		t.Fatal(err)
 	}
@@ -505,7 +514,8 @@ func TestHealingDanglingObject(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	fileInfoPostHeal, err = disks[0].ReadVersion(context.Background(), bucket, object, "", false)
+	disk = getDisk(0)
+	fileInfoPostHeal, err = disk.ReadVersion(context.Background(), bucket, object, "", false)
 	if err != nil {
 		t.Fatal(err)
 	}

--- a/cmd/erasure-healing_test.go
+++ b/cmd/erasure-healing_test.go
@@ -450,17 +450,14 @@ func TestHealingDanglingObject(t *testing.T) {
 		t.Fatalf("Expected versions 1, got %d", fileInfoPreHeal.NumVersions)
 	}
 
-	ch := make(chan struct{})
 	if err = objLayer.HealObjects(ctx, bucket, "", madmin.HealOpts{Remove: true},
 		func(bucket, object, vid string) error {
 			_, err := objLayer.HealObject(ctx, bucket, object, vid, madmin.HealOpts{Remove: true})
-			close(ch)
 			return err
 		}); err != nil {
 		t.Fatal(err)
 	}
 
-	<-ch
 	fileInfoPostHeal, err = disks[0].ReadVersion(context.Background(), bucket, object, "", false)
 	if err != nil {
 		t.Fatal(err)

